plan/plan.go

```diff
@@ -115,13 +131,18 @@ func (p *requiredProp) hashCode() []byte {
 	if p.hashcode != nil {
 		return p.hashcode
 	}
-	hashcodeSize := 8 + 8 + 8 + 16*len(p.cols)
+	hashcodeSize := 8 + 8 + 8 + 16*len(p.cols) + 8
 	p.hashcode = make([]byte, 0, hashcodeSize)
 	if p.desc {
 		p.hashcode = codec.EncodeInt(p.hashcode, 1)
 	} else {
 		p.hashcode = codec.EncodeInt(p.hashcode, 0)
 	}
+	if p.enforced {
```

Comment:
I think it should consider `enforced`. In `findBestTask` we can find this fact, when some `prop.cols != nil`, `findBestTask` can't generate any task to fit the `prop`, but `enforced == true` will generate the best task with `prop.cols = nil` and attach the task to `physicalSort`. If hash code don't consider `enforced`, **next time** we `findBestTask` with **same prop but `prop.enforced` is false**, we check the map, and will return the task whose `prop.enforced` is true.
The above situation cause wrong task.

Reply:
Yes, you are right, but it just looks weird to me. If there is an enforced task with less cost, but we cannot use it because we do not want to enforce it, so we choose a task with larger cost... Maybe we can refine it later.